### PR TITLE
feat(giveaway): added giveaway-end command

### LINKED ISSUE #1811: request: Additional sub commands for some modules.
### AFK Command/Module

Add other sub commands to the AFK command or make AFK command a separate module.
- [x] **Ignored Channels:** a command to add channel as blacklisted.
- [x] **Allowed Role List:** only these roles can use the command other than mods.
- [x] **Reset:** to set yours/someone's AFK reason to default.
- [x] **Clear:** removes the AFK status of a member.
- [x] **List:** list of members who have set AFK status.

### Giveaway

- [x] End: command to end the ongoing giveaway whenever mods what to.

#### Thanks for reading and you might implement these changes/updates to Skyra someday. <3

### PR DESCRIPTION
**Pull request by kyranet**
Fixes #1811

### CODE REVIEW

File: typescript/src/commands/Giveaway/gend.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { kRawEmoji } from '#lib/database';
+import { LanguageKeys } from '#lib/i18n/languageKeys';
+import { SkyraCommand } from '#lib/structures';
+import type { GuildMessage } from '#lib/types';
+import { PermissionLevels } from '#lib/types/Enums';
+import { ApplyOptions } from '@sapphire/decorators';
+import { Message } from 'discord.js';
+
+@ApplyOptions<SkyraCommand.Options>({
+	aliases: ['ge', 'giveaway-end'],
```
Comment (XynoxTheDev) [inline typescript/src/commands/Giveaway/gend.ts:10]: The alias can also be `gend` itself.

Comment (XynoxTheDev) [inline typescript/src/commands/Giveaway/gend.ts:10]: The alias can also be `gend` itself.

File: typescript/src/languages/en-US/commands/giveaway.json [hunk 2]
```diff
@@ -101,5 +101,36 @@
 			"30m 6h A hug from Skyra."
 		]
 	},
-	"giveawayRerollInvalid": "The message ID does not exist or there is no finished giveaway."
-}
\ No newline at end of file
+	"giveawayRerollInvalid": "The message ID does not exist or there is no finished giveaway.",
+	"giveawayEndDescription": "Ends a running giveaway message.",
```
Comment (XynoxTheDev) [inline typescript/src/languages/en-US/commands/giveaway.json:105]: Format for Giveaway End could be :point_down:
```
:tada: Congrats <winner>! You won <Prize> 
<message_link> 
```
(Message link of the ended giveaway in next line.)

**Review by XynoxTheDev — CHANGES_REQUESTED** [review]
fix(giveaway): Alias and End response.

File: typescript/src/commands/Giveaway/gend.ts (hunk 1 above)
Comment (kyranet) [reply typescript/src/commands/Giveaway/gend.ts:10]: `gend` itself is included in the command name.

File: typescript/src/languages/en-US/commands/giveaway.json (hunk 2 above)
Comment (kyranet) [reply typescript/src/languages/en-US/commands/giveaway.json:105]: No, it ends the giveaway, which makes it post this response, we don't need to duplicate this code.

File: typescript/src/languages/en-US/commands/giveaway.json [hunk 3]
```diff
@@ -101,5 +101,36 @@
...
+		],
+		"examples": [
+			"#giveaways 739417065466888262",
+			"739417065466888262",
+			"https://discord.com/channels/254360814063058944/739242237489315869/739417065466888261"
+		]
+	},
+	"giveawayEndMessageNotMine": "I am sorry, but that does not seem like a giveaway that is being run by me!",
+	"giveawayEndMessageNotActive": "The message you have given me looks like it can be a giveaway, but I am not tracking it. If you want to re-roll a finished giveaway, please run `greroll`.",
+	"giveawayEnd": "Successfully finished the giveaway at {{url}}!"
```
Comment (Favna) [inline typescript/src/languages/en-US/commands/giveaway.json]:
```suggestion
	"giveawayEnd": "Successfully finished the giveaway at <{{url}}>!"
```

**Review by Favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**